fastscanner/services/scanners/lib/parabolic.py

```diff
@@ -102,3 +103,68 @@ async def scan(
         df = df[(df[cum_low.column_name()] - df[C.LOW]).abs() < 0.0001]
 
         return df
+
+
+class ATRParabolicUpScanner:
+    def __init__(
+        self,
+        min_adv: float,
+        min_adr: float,
+        atr_multiplier: float,
+        min_volume: float,
+    ) -> None:
+        self._min_adv = min_adv
+        self._min_adr = min_adr
+        self._atr_multiplier = atr_multiplier
+        self._min_volume = min_volume
+
+    async def scan(
+        self, symbol: str, start: date, end: date, freq: str
+    ) -> pd.DataFrame:
+        adv = ADVIndicator(period=14)
+        adr = ADRIndicator(period=14)
+        atr = ATRIndicator(period=14, freq=freq)
+
+        daily_df = await ApplicationRegistry.candles.get(symbol, start, end, "1d")
+        if daily_df.empty:
+            return daily_df
+
+        daily_df = await adv.extend(symbol, daily_df)
+        daily_df = await adr.extend(symbol, daily_df)
+
+        daily_df = daily_df[daily_df[adv.column_name()] >= self._min_adv]
+        daily_df = daily_df[daily_df[adr.column_name()] >= self._min_adr]
+        if daily_df.empty:
+            return daily_df
+
+        df = await ApplicationRegistry.candles.get(symbol, start, end, freq)
+        df = df.loc[df.index.time <= time(9, 30)]  # type: ignore
+        if df.empty:
+            return df
+
+        df["date"] = df.index.date
+        daily_df = daily_df.set_index(daily_df.index.date)[[adv.column_name(), adr.column_name()]]  # type: ignore
+        df = df.join(daily_df, on="date", how="inner")
+        df = df.drop(columns=["date"])
+
+        if df.empty:
+            return df
+
+        df = await atr.extend(symbol, df)
+
+        cum_high = CumulativeIndicator(C.HIGH, CumOp.MAX)
+        df = await cum_high.extend(symbol, df)
+        cum_volume = CumulativeDailyVolumeIndicator()
+        df = await cum_volume.extend(symbol, df)
+
+        df = df[
+            (df[C.CLOSE] - df[C.OPEN]) > df[atr.column_name()] * self._atr_multiplier
+        ]
+
+        df = df[df[C.CLOSE] > df[C.OPEN]]
```

Comment:
I just realized that we don't need this condition because the one above (between line 160 and 162) already contains this one.